File: src/main/java/com/pubnub/api/models/server/access_manager/AccessManagerGrantPayload.java
package com.pubnub.api.models.server.access_manager;

import com.google.gson.JsonElement;
import com.google.gson.annotations.SerializedName;
import com.pubnub.api.models.consumer.access_manager.PNAccessManagerKeyData;
import com.pubnub.api.models.consumer.access_manager.PNAccessManagerKeysData;
import lombok.Getter;

import java.util.Map;

@Getter
public class AccessManagerGrantPayload {

    @SerializedName("level")
    private String level;

    private int ttl;

    @SerializedName("subscribe_key")
    private String subscribeKey;

    @SerializedName("channels")
    private Map<String, PNAccessManagerKeysData> channels;

    @SerializedName("channel-groups")
    private JsonElement channelGroups;

    @SerializedName("auths")
    private Map<String, PNAccessManagerKeyData> authKeys;

    @SerializedName("channel")
    private String channel;

}

File: src/main/kotlin/com/pubnub/api/models/server/access_manager/AccessManagerGrantPayload.kt
package com.pubnub.api.models.server.access_manager

import com.google.gson.JsonElement
import com.google.gson.annotations.SerializedName
import com.pubnub.api.models.consumer.access_manager.PNAccessManagerKeyData
import com.pubnub.api.models.consumer.access_manager.PNAccessManagerKeysData

class AccessManagerGrantPayload {

    internal lateinit var level: String

    internal var ttl = 0

    @SerializedName("subscribe_key")
    internal lateinit var subscribeKey: String

    internal val channels: Map<String, PNAccessManagerKeysData>? = null

    @SerializedName("channel-groups")
    internal val channelGroups: JsonElement? = null

    @SerializedName("auths")
    internal val authKeys: Map<String, PNAccessManagerKeyData>? = null

    internal val channel: String? = null
}